### PR TITLE
[REF] Consolidate code in processMembership

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Code cleanup

Now we have test cover from https://github.com/civicrm/civicrm-core/pull/17605 and other
recents we can consolidate this code & be comfortable the test cover will pick up issues.

Before
----------------------------------------
Duplicated & unnecessary code 

After
----------------------------------------
33 less lines of it!

Technical Details
----------------------------------------
I added test cover here before touching it!

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Member/Form/MembershipRenewal.php [hunk 1]
```diff
@@ -787,54 +785,29 @@ public function processMembership($contactID, $membershipTypeID, $is_test, $chan
     // Check and fix the membership if it is STALE
     CRM_Member_BAO_Membership::fixMembershipStatusBeforeRenew($currentMembership, $changeToday);
 
+    // CRM-7297 Membership Upsell - calculate dates based on new membership type
```
Comment (eileenmcnaughton) [inline CRM/Member/Form/MembershipRenewal.php:788]: All of these have been moved outside the if below as they are the same in both places

File: CRM/Member/Form/MembershipRenewal.php [hunk 2]
```diff
@@ -787,54 +785,29 @@ public function processMembership($contactID, $membershipTypeID, $is_test, $chan
...
-      $memParams = [];
-      $memParams['join_date'] = $membership->join_date;
-      $memParams['start_date'] = $membership->start_date;
-      $memParams['end_date'] = $dates['end_date'] ?? NULL;
-      $memParams['membership_type_id'] = $membershipTypeID;
-
-      //set the log start date.
-      $memParams['log_start_date'] = CRM_Utils_Date::customFormat($dates['log_start_date'], $format);
-
-      if (empty($membership->source)) {
```
Comment (eileenmcnaughton) [inline CRM/Member/Form/MembershipRenewal.php:831]: gone - tests show unneeded

File: CRM/Member/Form/MembershipRenewal.php [hunk 3]
```diff
@@ -787,54 +785,29 @@ public function processMembership($contactID, $membershipTypeID, $is_test, $chan
...
-
-      // Insert renewed dates for CURRENT membership
-      $memParams = [];
-      $memParams['join_date'] = $membership->join_date;
-      $memParams['start_date'] = $membership->start_date;
-      $memParams['end_date'] = $dates['end_date'] ?? NULL;
-      $memParams['membership_type_id'] = $membershipTypeID;
-
-      //set the log start date.
-      $memParams['log_start_date'] = CRM_Utils_Date::customFormat($dates['log_start_date'], $format);
```
Comment (eileenmcnaughton) [inline CRM/Member/Form/MembershipRenewal.php:829]: gone - tests show unneeded

File: CRM/Member/Form/MembershipRenewal.php [hunk 4]
```diff
@@ -787,54 +785,29 @@ public function processMembership($contactID, $membershipTypeID, $is_test, $chan
...
-      // CRM-7297 Membership Upsell - calculate dates based on new membership type
-      $dates = CRM_Member_BAO_MembershipType::getRenewalDatesForMembershipType($membership->id,
-        $changeToday,
-        $membershipTypeID,
-        $numRenewTerms
-      );
-
-      // Insert renewed dates for CURRENT membership
-      $memParams = [];
-      $memParams['join_date'] = $membership->join_date;
```
Comment (eileenmcnaughton) [inline CRM/Member/Form/MembershipRenewal.php:823]: gone - tests show unneeded

File: CRM/Member/Form/MembershipRenewal.php [hunk 5]
```diff
@@ -787,54 +785,29 @@ public function processMembership($contactID, $membershipTypeID, $is_test, $chan
...
-        $changeToday,
-        $membershipTypeID,
-        $numRenewTerms
-      );
-
-      foreach (['start_date', 'end_date'] as $dateType) {
-        $currentMembership[$dateType] = $dates[$dateType] ?? NULL;
-      }
-      $currentMembership['is_test'] = $is_test;
-      $memParams = $currentMembership;
```
Comment (eileenmcnaughton) [inline CRM/Member/Form/MembershipRenewal.php:805]: gone - tests show unneeded

File: CRM/Member/Form/MembershipRenewal.php [hunk 6]
```diff
@@ -847,12 +820,6 @@ public function processMembership($contactID, $membershipTypeID, $is_test, $chan
       $memParams['contribution_recur_id'] = $contributionRecurID;
     }
 
-    //since we are renewing,
-    //make status override false.
-    $memParams['is_override'] = FALSE;
```
Comment (eileenmcnaughton) [inline CRM/Member/Form/MembershipRenewal.php:852]: moved up

**Comment by seamuslee001** [comment]
Test fail related

**Comment by seamuslee001** [comment]
@eileenmcnaughton `PHP Parse error:  syntax error, unexpected '=', expecting ']' in CRM/Member/Form/MembershipRenewal.php on line 798`

**Comment by mlutfy** [comment]
Seems good:

* Did a code review, resulting code is more clear
* Tests are passing

**Comment by seamuslee001** [comment]
This has been merge ready for a couple of days now, no one has objected merging

**Pull request merged**